simplecpp.cpp

```diff
@@ -1839,8 +1839,37 @@ namespace simplecpp {
 }
 
 #ifdef SIMPLECPP_WINDOWS
+
+class RealFileNameMap
+{
+public:
+    bool getRealPathFromCache(std::string const& path, std::string& returnPath)
+    {
+        std::unique_lock<std::mutex> lock(mMutex);
```

Comment:
In this project we still use C++03 to have maximum portability. And `std::unique_lock` is a C++11 feature.

I would prefer to use the windows function to lock instead of `std::unique_lock`. Can you please try this and if it does not work well.. let me know.

I guess you can use the functions described here:
https://docs.microsoft.com/en-us/windows/desktop/sync/using-mutex-objects